.github/workflows/build.yml

```diff
@@ -0,0 +1,170 @@
+  deploy_test:
+    name: Verify PyPI deployment confirmation
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_REF: ${{ github.ref }}
+      CONFIRM_REF: ${{ github.event.inputs.confirm_ref }}
+    steps:
+      - name: Compare confirmation to current reference
+        shell: bash
+        run: |
+          [[ -z $CONFIRM_REF || $GITHUB_REF =~ ^refs/(heads|tags)/$CONFIRM_REF$ ]]
+          if [[ -z $CONFIRM_REF ]]; then
+            echo "Build only.  Nothing will be uploaded to PyPI."
+          else
+            echo "Full build and deploy.  Wheels and source will be uploaded to PyPI."
+          fi
+
+
+  build_sdist:
+    name: Build sdist on Ubuntu
+    needs: deploy_test
+    runs-on: ubuntu-latest
+    env:
+      OVERRIDE_VERSION: ${{ github.event.inputs.override_version }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          # For the sdist we should be as conservative as possible with our
+          # Python version.  This should be the lowest supported version.  This
+          # means that no unsupported syntax can sneak through.
+          python-version: '3.6'
+
+      - name: Install pip build
+        run: |
+          python -m pip install 'build'
+
+      - name: Build sdist tarball
+        shell: bash
+        run: |
+          if [[ ! -z "$OVERRIDE_VERSION" ]]; then echo "$OVERRIDE_VERSION" > VERSION; fi
+          # The build package is the reference PEP 517 package builder.  All
+          # dependencies are specified by our setup code.
+          python -m build --sdist .
+
+      # Zip files are not part of PEP 517, so we need to make our own.
+      - name: Create zipfile from tarball
+        shell: bash
+        working-directory: dist
+        run: |
+          # First assert that there is exactly one tarball, and find its name.
+          shopt -s failglob
+          tarball_pattern="*.tar.gz"
+          tarballs=($tarball_pattern)
+          [[ ${#tarballs[@]} == 1 ]]
+          tarball="${tarballs[0]}"
+          # Get the stem and make the zipfile name.
+          stem="${tarball%.tar.gz}"
+          zipfile="${stem}.zip"
+          # Extract the tarball and rezip it.
+          tar -xzf "$tarball"
+          zip "$zipfile" -r "$stem"
+          rm -r "$stem"
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: sdist
+          path: |
+            dist/*.tar.gz
+            dist/*.zip
+          if-no-files-found: error
+
+
+  build_wheels:
+    name: Build wheels
+    needs: deploy_test
+    runs-on: ubuntu-latest
+    env:
+      OVERRIDE_VERSION: ${{ github.event.inputs.override_version }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          # This is about the build environment, not the released wheel version.
+          python-version: '3.7'
```

Comment:
Notionally the same argument about minimum Python version applies here as well as on the sdist (i.e. you want to build using the minimum supported version), but _really_ I don't think it matters at all as long as it's done with the lowest in at least one place.

Reply:
(the reason it doesn't apply here for qutip/qutip is that the build is delegated to cibuildwheel, which then sets up a bunch of docker containers or something with each supported Python version)